### PR TITLE
Improve error when running snippet that requires --write-locks

### PR DESCRIPTION
**Pull request by big-guy**
inspired from #25294

### CODE REVIEW

**Comment by big-guy** [comment]
@bot-gradle test and merge

File: subprojects/docs/src/snippets/dependencyManagement/dependencyLocking-lockingAllConfigurations/kotlin/build.gradle.kts [hunk 1]
```diff
@@ -12,7 +12,7 @@ dependencyLocking {
 tasks.register("resolveAndLockAll") {
     notCompatibleWithConfigurationCache("Filters configurations at execution time")
     doFirst {
-        require(gradle.startParameter.isWriteDependencyLocks)
+        require(gradle.startParameter.isWriteDependencyLocks, "$path must be run from the command line with the `--write-locks` flag")
```
Comment (aSemy) [inline subprojects/docs/src/snippets/dependencyManagement/dependencyLocking-lockingAllConfigurations/kotlin/build.gradle.kts:15]: [`require()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/require.html) from the Kotlin stdlib uses a lazy message

```suggestion
        require(gradle.startParameter.isWriteDependencyLocks) { "$path must be run from the command line with the `--write-locks` flag") }
```

Comment (big-guy) [reply subprojects/docs/src/snippets/dependencyManagement/dependencyLocking-lockingAllConfigurations/kotlin/build.gradle.kts:15]: 🤦 yes of course

**Comment by big-guy** [comment]
@bot-gradle test and merge

**Pull request merged**